int32 lower
int32 upper
---
int32 errorCode
string errorDescription